### DISCUSSION TITLE
cobbler boot menu

### DISCUSSION THREAD

**Discussion by 1392273211** [cobbler/cobbler#3630] [Q&A]
![1](https://github.com/cobbler/cobbler/assets/53630341/636cf99c-db32-4fe1-a102-b81786b84253)
Cobbler boot menu, for example, I have centos7.6 7.9 8.0 8.5. The client starts with pxe to enter the menu of the boot installation system. Assuming that the default boot installation is 7.6, how can I make 7.9 become the default boot installation option?

**Comment by SchoolGuy** [comment]
What you need to do is grab the MAC address of the system before installation, perform a `cobbler system add --name="example" --ip-address="<your ip of choice>" --mac-address="AA:BB:CC:DD:EE:FF" --profile="centos7.9-x86_64" --netboot-enabled="true" ...`. That will put the entry of your choice as the default entry for this system.

**Reply by 1392273211** [reply]
That is, after I configured Distros and Profiles, I configured centos7.6 7.6 7.8 7.7. Then there will be multiple options when starting the server. How can I control which option I want to install through the configuration file instead of selecting it through the up and down keyboard?

**Reply by SchoolGuy** [reply]
As I wrote you do this through adding a Cobbler System for the target machine. If the flag `netboot-enabled` is set to `True` then the system will be booted. During the install process, this is set to false which then means the local disk is booted after the installation.

**Reply by 1392273211** [reply]
I understand what you mean. What I want to know is, if I don’t use the systems button, but only use profiles and distros to control the options on the pxe interface to decide which operating system to install, otherwise I use systems to install, then I have different models of servers with different network card names. I cannot use the same set of ks files. Then I need to create multiple list information in the systems list, so I want to use profiles and distros to control the options on the pxe interface. I don’t know about cobbler. Is there this function to control

**Reply by SchoolGuy** [reply]
Well no this is not possible. You can create a system named default that controls a single profile that is booted per default but if you have multiple different groups of systems then this won't work. I strongly recommend using Cobbler Systems as this is the only way to guarantee that the intended entry is booted for a heterogeneous group of hardware.

**Comment by NileshChandekar** [comment]
@1392273211 just try and change the boot order in ``/var/lib/tftpboot/pxelinux.cfg/default``

**Reply by 1392273211** [reply]
Thank you, I have tried this method, but there is only one configuration file left in it, and there is no way to dynamically modify the message.

**Reply by NileshChandekar** [reply]
@1392273211 I don't think so there is a way to change this dynamically. NO more Idea.
